DuckDuckGo/NetworkProtectionStatusViewModel.swift

```diff
@@ -63,17 +73,34 @@ final class NetworkProtectionStatusViewModel: ObservableObject {
             .map(Self.statusImageID(connected:))
             .assign(to: \.statusImageID, onWeaklyHeld: self)
             .store(in: &cancellables)
+    }
 
+    private func setUpStatusMessagePublishers() {
         statusObserver.publisher
-            .map { $0.isLoading }
+            .flatMap(maxPublishers: .max(1)) { status in
+                // As soon as the connection status changes, we should update the status message
+                var statusUpdatePublishers = [Just(Self.message(for: status)).eraseToAnyPublisher()]
+                switch status {
+                case .connected(let connectedDate):
+                    // In the case that the status is connected, we should then provide timed updates
+                    // If we rely on the timed updates alone, there will be a delay to the initial update
+                    statusUpdatePublishers.append(Self.timedConnectedStatusMessagePublisher(forConnectedDate: connectedDate))
+                default:
+                    break
+                }
+                return statusUpdatePublishers.publisher
+            }
```

Comment:
Interesting.

Combine can be complicated at times, but the code makes sense and uses a nice approach to the issue at hand.  Nice work.